### ISSUE TITLE
RuboCop ignores .ruby-version if it specifies a Ruby patch version

### ISSUE THREAD

**Issue by andersennl** [bbatsov/rubocop#4103]
Hi, I've noticed the RuboCop linter showed an error when using the safe navigating operator, as the Ruby version used for linting didn't include this functionality. After I read that RuboCop uses the Ruby version specified in `.ruby-version` ([link](https://github.com/bbatsov/rubocop/blob/master/manual/configuration.md#setting-the-target-ruby-version)), I wondered why it actually didn't.

--------

## Expected behavior

I expect RuboCop to use the Ruby version specified in my `.ruby-version` file. It specifies `2.3.1`.

## Actual behavior

RuboCop seems to ignore this version and fall back to a much older version.

## Steps to reproduce the problem

I specified Ruby v`2.3.1` via `TargetRubyVersion` in `.rubocop.yml`, which gave me this error when I invoked RuboCop:

```ruby
Error: Unknown Ruby version "2.3.1" found in `TargetRubyVersion` parameter (in /Users/nandersen/Projects/myproject/.rubocop.yml).
Known versions: 1.9, 2.0, 2.1, 2.2, 2.3, 2.4
```

Coming back to the expected behavior: I would expect RuboCop to fall back to / use to the highest available minor version, below the specified patch version. At least it would be nice if it did, otherwise one would have to both update `.ruby-version` and `.rubocop.yml` when updating the Ruby version.

## RuboCop version
```
$ rubocop -V
0.47.1 (using Parser 2.3.3.1, running on ruby 2.3.1 x86_64-darwin15)
```

**Comment by smakagon** [comment]
@bbatsov @pocke @Drenmi It's easy to add support of patch version to rubocop. All we need to do is change this line:
https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/config.rb#L288
From:
`/\A(ruby-)?(?<version>\d+\.\d+)/ =~ File.read('.ruby-version')`
To:
`/\A(ruby-)?(?<version>\d+\.\d+(.\d+)?)/ =~ File.read('.ruby-version')`

But in that method we have: `version.to_f`
And entire Config class uses ruby versions as floats. For example here:
https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/config.rb#L20

Is it safe to convert ruby version from float to string? Is there any particular reason why float was used?

If we can use strings I could change that class.

**Comment by pocke** [comment]
I guess we can strip patch level, because Ruby uses semantic versionsing.
https://www.ruby-lang.org/en/news/2013/12/21/ruby-version-policy-changes-with-2-1-0/

**Comment by smakagon** [comment]
I was thinking about something like that:
https://github.com/smakagon/rubocop/commit/a6ab11b1472523efa4c16f6a5a98ecb2b64f75ab
We can make sure that we have ruby version as a float (1.9, 2.3, etc) when we load config file.

My first idea was to convert string to float inside `Config` class, but it's better to convert string to float as early as possible and have consistent type.

With my changes:
If we specify TargetRubyVersion for AllCops as 2.3.3 - ConfigLoader will parse it as 2.3 (float). 
By default it would return it as a string "2.3.3" (that's what causes an issue now)

**Comment by bbatsov** [comment]
I'm fine either way.

**Comment by tadeuasarro** [comment]
I had the same issue when creating a PR, for the .github/workflows specified an older version of Rubocop, just updated that to the latest one and it's all fixed! :D
